src/include/migraphx/shape.hpp

```diff
@@ -381,6 +383,8 @@ struct MIGRAPHX_EXPORT shape
     {
         switch(t)
         {
+        case uint4_type:
+        case int4_type:
```

Comment:
The tuple visitor should not be called for these types. It should throw an error.